### ISSUE TITLE
limit task interaction to admins

### ISSUE THREAD

**Issue by JonasJurczok** [JonasJurczok/factorio-todo-list#72]
As server admin
I would like to limit task interaction to certain users
So that not every troll can flood the system with BS tasks.

Acceptance Criteria:
* It is possible to limit task interaction
* Only people with the correct permissions can create/assign/edit/delete tasks

Original request [here](https://mods.factorio.com/mod/Todo-List/discussion/5bd823b6b37261000cd1c590)

**Comment by JonasJurczok** [comment]
This should actually be rather easy to implement.
See 
* https://lua-api.factorio.com/latest/LuaPermissionGroup.html
* https://lua-api.factorio.com/latest/LuaPermissionGroups.html

**Comment by JonasJurczok** [comment]
@Tarrke Check the two links :)

**Comment by JonasJurczok** [comment]
As I understood it the UI is already there.
We have two options according to the API

Either we create our own role management (rather tricky)

Or we add a couple options that define group names.
If groups of that name exists then only players in that group can do the action.

The latter one should be rather straight forward.

**Comment by Tarrke** [comment]
Okay found the UI in question in MP or SP game open the console and hit
```lua
/permissions
```

Now I see things this should be simple yeah.

**Comment by Tarrke** [comment]
After some researche it does not seems possible to add new input_action rules. See https://forums.factorio.com/viewtopic.php?f=221&t=47308

We should be able to link the right to add/edit/assign tasks to an existing rule, but which one ???
List of defined onput_action here : https://lua-api.factorio.com/latest/defines.html

**Comment by Tarrke** [comment]
I'm almost positive that the `defines.input_action.add_permission_group` may be used here.

**Comment by JonasJurczok** [comment]
Yep. But this just enables us to add new permission groups. Not permissions :)

**Comment by Tarrke** [comment]
We can add new permissions, but we can base the permission system in the mod on existing permissions. The OP asked for "only admins" functionnality, I think we can give the players with the add_permission_group permission to manage tasks. This is a workaround.

**Comment by JonasJurczok** [comment]
Yeah, that would work as well.

**Comment by Tarrke** [comment]
Should this be an option as well ?

**Comment by JonasJurczok** [comment]
I would say it has to. Otherwise you force everyone into this limitation :)

**Comment by Tarrke** [comment]
I'll look into that next week

**Comment by JonasJurczok** [comment]
Okay just to recap, there are two ways of doing this.

1) Bind the rights to the `defines.input_action.add_permission_group` permission
Question:
There are multiple permissions in this mod (create, edit, assign, complete, delete). How are they distinguished (you want completion of tasks for everyone but not creation?)

2) Create a set of options (one per functionality).
These options are free form text.
If they are set it is checked if a group of that name exists.
If a group exists it is checked if the user is in the group.
If yes the action is performed.

**Comment by JonasJurczok** [comment]
As a side effect: would it make sense to have two levels of permissions? One for doing everything and one for only working with tasks assigned to yourself?
